boards: arm: add support for stm32f072b_disco board

dwagenk opened the pull request.

Commit:
boards: arm: add support for STM32F072B-DISCO

Support the ST STM32F072B-DISCO discovery board with
STM32F072RB SoC

Signed-off-by: Daniel Wagenknecht <wagenknecht@clage.de>

DWagenk commented:
shippable returns unstable with linking errors in some tests (e.g. tests/net/ieee802154/l2/test, tests/net/lib/mqtt_subscriber/test), because of small SRAM. I expected 
`ram: 16`
at [stm32f072b_disco.yaml](https://github.com/DWagenk/zephyr/blob/29d411947b13ba31470e0938e1df98114c025bd1/boards/arm/stm32f072b_disco/stm32f072b_disco.yaml#L8) to prevent the tests beeing run for this target.

How should this be solved?

galak commented:
> shippable returns unstable with linking errors in some tests (e.g. tests/net/ieee802154/l2/test, tests/net/lib/mqtt_subscriber/test), because of small SRAM. I expected
> ram: 16
> at stm32f072b_disco.yaml to prevent the tests beeing run for this target.
> 
> How should this be solved?

For most of the smaller ram boards we've done something like the following in BOARD.yaml
```
testing:
  ignore_tags:
    - net
    - bluetooth
```

See boards/arm/frdm_kl25z/frdm_kl25z.yaml as an example

dbkinder commented on boards/arm/stm32f072b_disco/doc/stm32f072b_disco.rst:
```diff
@@ -0,0 +1,216 @@
+
+- Six LEDs:
+
+       - LD1 (red/green) for USB communication
```
This sublist looks better when it's not indented so much, 4 spaces is sufficient.  Otherwise the html that's generated has a vertical bar added on the left because it looks like a quoted text block.

DWagenk replied:
done

DWagenk replied:
actually I had to go down to 2 Spaces to remove the quote-bar.

dbkinder reviewed (changes requested):
One formatting issues

DWagenk commented:
fixup:
I missed, that doc and pinmux.c were out of sync. 
Added I2C_1 to pinmux.c and removed I2C_2 from doc.

erwango reviewed (commented):
SoC STM32F072xB is being introduced as part of #5171 which should be close to merge.
I think you should wait it is merged and rebase on top of it.
Otherwise LGTM

superna9999 commented:
Yep, I think also, you also should enable FLASH in the defconfig.

Apart that, LGTM

galak commented:
Can you rebase this?

Commit:
pinmux: stm32f0: add additional defines for I2C2

Adds pinmux defines to use I2C2 at PB10/PB11 for
stm32f0-based boards.
Needed for stm32f072b_disco board to use extension
connector

Signed-off-by: Daniel Wagenknecht <wagenknecht@clage.de>

DWagenk commented:
rebased onto #5171 that introduce stm32f072xb SoC.

@superna9999 I removed FLASH from the documentation, since I didn't test it at.

DWagenk commented:
recheck

DWagenk commented:
do not merge

I had some issues with the uart configuration right now and want to look into it tomorrow

DWagenk commented:
now that #5171 is merged this should be ready to get merged as well.

@ydamigos can you please take a look at the SPI configuration through device tree? SPI_loopback test works, but I'm not familiar with dts, so I stay a little uncertain, if all is correct. I basically copied your work from 96b_carbon board and adjustet the dts.fixup file to use the SPI_FIFO versions instead.

@superna9999 @erwango @dbkinder can you please take a look at it again?

ydamigos commented on tests/drivers/spi/spi_loopback/prj_stm32f072b_disco.conf:
```diff
@@ -0,0 +1,10 @@
+CONFIG_BUILD_TIMESTAMP=y
+CONFIG_SYS_LOG=y
+CONFIG_SPI=y
+CONFIG_SPI_LEGACY_API=n
```
It isn't needed. It defaults to n in arm/soc/st_stm32/common/Kconfig.defconfig.series when SPI and one SPI_x port are enabled.

ydamigos commented on tests/drivers/spi/spi_loopback/prj_stm32f072b_disco.conf:
```diff
@@ -0,0 +1,10 @@
+CONFIG_SPI=y
+CONFIG_SPI_LEGACY_API=n
+CONFIG_SYS_LOG_SPI_LEVEL=1
+CONFIG_SPI_STM32=y
```
It isn't needed. It defaults to y in arm/soc/st_stm32/common/Kconfig.defconfig.series when SPI and one SPI_x port are enabled.

ydamigos commented on tests/drivers/spi/spi_loopback/prj_stm32f072b_disco.conf:
```diff
@@ -0,0 +1,10 @@
+CONFIG_SPI_LEGACY_API=n
+CONFIG_SYS_LOG_SPI_LEVEL=1
+CONFIG_SPI_STM32=y
+CONFIG_SPI_STM32_INTERRUPT=y
```
It isn't needed. It defaults to y in boards Kconfig.defconfig when SPI is enabled.

ydamigos commented on tests/drivers/spi/spi_loopback/prj_stm32f072b_disco.conf:
```diff
@@ -0,0 +1,10 @@
+CONFIG_SYS_LOG_SPI_LEVEL=1
+CONFIG_SPI_STM32=y
+CONFIG_SPI_STM32_INTERRUPT=y
+CONFIG_SPI_1=y
```
It isn't needed. It defaults to y in boards Kconfig.defconfig when SPI is enabled.

ydamigos reviewed (commented):
LGTM. Some minor non blocking comments. I don't have the hardware to test it.

erwango commented:
@DWagenk : Approved, but please take into account @ydamigos comments

DWagenk commented:
@erwango @ydamigos what do you think of moving those changes to a seperate PR that cleans up the spi_loopback test?
It would effect also those files in tests\drivers\spi\spi_loopback:
- prj_96b_carbon_irq.conf
- prj_96b_carbon_polled.conf
- prj_nucleo_f091rc.conf (which doesn't support SPI right now anyhow, see #5836)
- prj_nucleo_l432kc.conf (which doesn't support SPI right now anyhow, see #5836)

erwango commented:
> what do you think of moving those changes to a seperate PR that cleans up the spi_loopback test?

@DWagenk : Sure, go ahead

DWagenk commented:
I won't have time for a follow up pr for the spi_loopback test before monday, but would like to get this closed soon (it's been open quite some time). Can you merge it now anyhow, or at least the two first commits (without adding the board to spi_loopback test)?

erwango commented:
> Can you merge it now anyhow, or at least the two first commits (without adding the board to spi_loopback test)?

@DWagenk: Maybe you can just remove last commit.

galak commented:
I pushed an update and dropped the SPI patch so we can merge this board port

DWagenk commented:
@galak thanks!
